### ISSUE TITLE
[BUG] First level .Include(...) doesnt matter

### ISSUE THREAD

**Issue by RedQueen87** [mbdavid/LiteDB#2533]
**Version**
I am using LiteDB 5.0.21 on Windows 11 and .NET 6.

**Describe the bug**
I am familiar with Entity Framework and when I query for one table I will receive data from this single table no matter how many references this table has. Unless I use .Include(...).
With LiteDB it is different and I am not sure if this is a bug or a feature.

**Code to Reproduce**
This is very easy example to reproduce.
Data structure is very simple.
-> Level1 collection
--> Level2 collection
---> Level3 collection
----> Level4 collection
Each collection refer to list of level n+1 collection except the last one (level 4)
Just paste this code to console application.
```
using LiteDB;
using System.Diagnostics;

Console.WriteLine("LiteDB DbRef Test");
Console.WriteLine("This example refer to https://www.litedb.org/docs/dbref/");

var filename = Path.GetTempFileName();
using (var db = new LevelDatabase(filename))
{
    Console.WriteLine("Fill with some dummy data");
    var l1 = new Level1 { Name = "Level 1" };
    var l2 = new Level2 { Name = "Level 2" };
    var l3 = new Level3 { Name = "Level 3" };
    var l4 = new Level4 { Name = "Level 4" };
    // relations
    l1.Children.Add(l2);
    l2.Children.Add(l3);
    l3.Children.Add(l4);
    // insert
    db.Level1.Insert(l1);
    db.Level2.Insert(l2);
    db.Level3.Insert(l3);
    db.Level4.Insert(l4);
}

Console.WriteLine("Stop here for a minute");
Console.WriteLine($"Open LiteDB Studio and preview this file: {filename}");
Debugger.Break();

Console.WriteLine("Reopen the same db file");
using (var db = new LevelDatabase(filename))
{
    Console.WriteLine();
    Console.WriteLine("TEST #1");
    Console.WriteLine("Lets query same data WITHOUT using include");
    var l1 = db
        .Level1
        .Query()
        // I do NOT include any level 1 Children property here
        .Single();
    Console.WriteLine($"How many children has l1? {l1.Children.Count} (WRONG?)");
    var l2 = l1.Children.Single();
    Console.WriteLine($"How many children has l2? {l2.Children.Count} (CORRECT)");
    Debugger.Break();

    Console.WriteLine();
    Console.WriteLine("TEST #2");
    Console.WriteLine("Lets do the same using include");
    l1 = db
        .Level1
        .Query()
        // I do include level 1 Children property here
        .Include(xl1 => xl1.Children)
        // but I do NOT include level 2 Children property here
        .Single();
    Console.WriteLine($"How many children has l1? {l1.Children.Count} (CORRECT)");
    l2 = l1.Children.Single();
    Console.WriteLine($"How many children has l2? {l2.Children.Count} (WRONG)");
    Debugger.Break();

    Console.WriteLine();
    Console.WriteLine("TEST #3");
    Console.WriteLine("Lets do the same using another include");
    l1 = db
        .Level1
        .Query()
        // I do include level 1 Children property here
        .Include(xl1 => xl1.Children)
        // and I do include level 2 Children property here
        .Include(xl1 => xl1.Children.Select(xl2 => xl2.Children))
        .Single();
    Console.WriteLine($"How many children has l1? {l1.Children.Count} (CORRECT)");
    l2 = l1.Children.Single();
    Console.WriteLine($"How many children has l2? {l2.Children.Count} (CORRECT)");
    var l3 = l2.Children.Single();
    Console.WriteLine($"How many children has l2? {l3.Children.Count} (WRONG)");
    Debugger.Break();

    Console.WriteLine();
    Console.WriteLine("TEST #4");
    Console.WriteLine("Lets make it even more weird");
    l1 = db
        .Level1
        .Query()
        // I do NOT include level 1 Children property here
        // and I do include level 2 Children property here
        .Include(xl1 => xl1.Children.Select(xl2 => xl2.Children))
        .Single();
    Console.WriteLine($"How many children has l1? {l1.Children.Count} (WRONG)");
    l2 = l1.Children.Single();
    Console.WriteLine($"How many children has l2? {l2.Children.Count} (WRONG)");
    Debugger.Break();

    Console.WriteLine("Conclusion:");
    Console.WriteLine("It doesn't matter if i declare .Include(...) or not on level 1 deep.");
    Console.WriteLine("It will always get level 2 children anyway.");

    Console.WriteLine();
    Console.WriteLine("And the question is:");
    Console.WriteLine("HOW CAN I GET Level 1 collection ONLY?");
    Debugger.Break();
}

Console.WriteLine("End of test");
Console.ReadKey();

#region Model

public class LevelDatabase : LiteDatabase
{
    public LevelDatabase(string connectionString)
        : base(connectionString)
    {
        this.OnCreateModel();
    }

    public ILiteCollection<Level1> Level1 => this.GetCollection<Level1>();
    public ILiteCollection<Level2> Level2 => this.GetCollection<Level2>();
    public ILiteCollection<Level3> Level3 => this.GetCollection<Level3>();
    public ILiteCollection<Level4> Level4 => this.GetCollection<Level4>();

    private void OnCreateModel()
    {
        this.Mapper
            .Entity<Level1>()
            .DbRef(x => x.Children);
        this.Mapper
            .Entity<Level2>()
            .DbRef(x => x.Children);
        this.Mapper
            .Entity<Level3>()
            .DbRef(x => x.Children);
    }
}

public class Level1
{
    [BsonId]
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; }
    public IList<Level2> Children { get; set; } = new List<Level2>();
}

public class Level2
{
    [BsonId]
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; }
    public IList<Level3> Children { get; set; } = new List<Level3>();
}

public class Level3
{
    [BsonId]
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; }
    public IList<Level4> Children { get; set; } = new List<Level4>();
}

public class Level4
{
    [BsonId]
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; }
}

#endregion
```

**Expected behavior**
In my opinion framework should give me full control over my database model.
So when I want to get data from ONLY one collection/table, I should be able to do it.
And based on the documentation [here](https://www.litedb.org/docs/dbref/) I understand this as expected.

**Comment by ssteiner** [comment]
Did you look at what is contained in these children? With DbRef, if you do not do the Include, you'll still get the Object, but it contains just the Id, nothing more. Then, to set a child, you don't have to look up the child, you just instantiate a new child with the right Id and LiteDB figures out the rest.
LiteDb stores object type and Id of a DbRef - and auto-hydrates it - but it doesn't really read the child, it just instantiates the object and fills the Id, and that's it.

Yes if feels weird coming from EF (I took the same path), and I had to adapt my workflow a bit, but in the end, it's just one of those things that are different between SQL and NoSql. I've been working with MongoDB and RavenDB recently and they have their own different way of handling these things - they don't auto-hydrate but their way of handling DbRefs is in the end even more different from EF (and not in a good way imho).
